src/screens/Settings/stacks/timer/Timer.tsx

```diff
@@ -0,0 +1,115 @@
+            dispatch(setSleepTime(value * 3600));
+        } else if (timeType == 'Minutes' && value > 0) {
+            dispatch(setSleepTime(value * 60));
+        }
+    }, [navigation, dispatch]);
+
+    const getTime = () => {
+        let hours = 0;
+        let minutes = 0;
+        if (!Number.isNaN(sleepTime)) {
+            hours = Math.round(sleepTime / 3600);
+            const timeRemaining = sleepTime % 3600;
+
+            if (timeRemaining >= 60) {
+                minutes = Math.round(timeRemaining / 60);
+            }
+        }
+
+        return `${hours} hrs ${minutes} min`;
+    }
+
+    const handleEnabledSleeper = useCallback((value: boolean) => {
+        dispatch(setEnabledSleeper(value));
+        setEnableSleeper(value);
+
+        // If value is true sleeper has been enabled, pause then play tack
+        // to trigger play state and start sleeper timer
+        if (value) {
+            TrackPlayer.pause()
+            TrackPlayer.play()
+        }
+    }, [dispatch]);
+
+    return (
+        <Container>
+            <InputContainer>
+                <Text>Set Sleep Timer. Time Set Previously: {getTime()}</Text>
+                <View style={timerStyles.checkbox}>
+                    <CheckBox
+                        value={enableSleeper}
+                        onValueChange={(value) => handleEnabledSleeper(value)}
+                    />
+                    <Text> Enable Sleeper</Text>
+                </View>
+                <View style={enabledSleeper ? timerStyles.timerSetting : timerStyles.timerSettingsDisabled}>
+                    <View style={timerStyles.timer}>
+                        <View style={timerStyles.timeInput}>
+                            <Text>H:</Text>
+                            <Input
+                                placeholder='60'
+                                editable={enabledSleeper}
+                                style={timerStyles.input}
+                                keyboardType='phone-pad'
+                                onChangeText={(value: string) => {
+                                    setSleeper(parseInt(value), 'Hours');
+                                }}
+                            />
+                        </View>
+                        <View style={timerStyles.timeInput}>
+                            <Text>M:</Text>
+                            <Input
+                                placeholder='60'
+                                editable={enabledSleeper}
+                                style={timerStyles.input}
+                                keyboardType='phone-pad'
+                                onChangeText={(value: string) => {
+                                    setSleeper(parseInt(value), 'Minutes');
+                                }}
+                            />
+                        </View>
+                    </View>
+                    <Text>Set this to automatically stop the audio when time runs out.</Text>
```

Comment:
There's an untranslated string here.

Reply:
What is the best approach leave the translation to be done after the PR or add now.